packages/dev/core/src/Materials/meshDebugPluginMaterial.ts

```diff
@@ -0,0 +1,514 @@
+        this._options.uvColor2 = Vector3.FromArray(serializationObject.uvColor2);
+        this._options.materialColor = Vector3.FromArray(serializationObject.materialColor);
+        this._options.distanceScale = serializationObject.distanceScale;
+        
+        this.markAllDefinesAsDirty();
+    }
+
+    /**
+     * Generate a random color
+     * @returns Random RGB color as a Vector3
+     */
+    private _getRandomColor(): Vector3 {
+        const hue: number = Math.random() * 360;
+        const color: Color3 = Color3.FromHSV(hue, 0.95, 0.99);
+        return Vector3.FromArray(color.asArray());
+    }
+
+    /**
+     * Renders triangles in a mesh 3 times by tripling the indices in the index buffer.
+     * Used to prepare a mesh to be rendered in `TRIANGLES`, `VERTICES`, or `TRIANGLES_VERTICES` modes.
+     * NOTE: This is a destructive operation. The mesh's index buffer and vertex buffers are modified, and a new vertex buffer is allocated.
+     * If you'd like the ability to revert these changes, toggle the optional `returnRollback` flag.
+     * @param mesh the mesh to target
+     * @param returnRollback whether or not to return a function that reverts mesh to its initial state. Default: false.
+     * @returns a rollback function if `returnRollback` is true, otherwise an empty function.
+     */
+    public static prepareMeshForTrianglesAndVerticesMode(mesh: Mesh, returnRollback: boolean = false): () => void {
+        let rollback = () => {};
+
+        if (mesh.getTotalIndices() == 0) return rollback;
+        
+        if (returnRollback) {
+            const kinds = mesh.getVerticesDataKinds();
+            const indices = mesh.getIndices()!;
+            const data: { [kind: string]: FloatArray } = {};
+            for(const kind of kinds) {
+                data[kind] = mesh.getVerticesData(kind)!;
+            }
+
+            rollback = function() {
+                mesh.setIndices(indices);
+                for(const kind of kinds) {
+                    mesh.setVerticesData(kind, data[kind]);
+                }
+                mesh.removeVerticesData("dbg_initialPass");
+            }
+        }
+
+        let indices = Array.from(mesh.getIndices()!);
+        const newIndices1 = [];
+        for (let i = 0; i < indices.length; i += 3) {
+            newIndices1.push(indices[i + 1], indices[i + 2], indices[i + 0]);
+        }
+        mesh.setIndices(indices.concat(newIndices1));
+
+        mesh.convertToUnIndexedMesh();
+
+        mesh.isUnIndexed = false;
+
+        indices = Array.from(mesh.getIndices()!);
+        const newIndices2 = [];
+        for (let i = indices.length / 2; i < indices.length; i += 3) {
+            newIndices2.push(indices[i + 1], indices[i + 2], indices[i + 0]);
+        }
+        mesh.setIndices(indices.concat(newIndices2));
+
+        const totalVertices = mesh.getTotalVertices() / 2;
+        const pass = new Array(totalVertices).fill(1).concat(new Array(totalVertices).fill(0));

```

Comment:
This should reduce allocations.
```suggestion
        const num = mesh.getTotalVertices();
        const mid = num / 2;
        const pass = new Array(num).fill(1, 0, mid).fill(0, mid, num);
```